draft-ietf-ace-oscore-profile.xml

```diff
@@ -715,57 +742,54 @@ OSCORE_Security_Context = {
+token.
         </t>
 
         <t>
-          If the exchange was an update of access rights, i.e. a new Security Context was derived from a client that already had a Security Context in place, the is RECOMMENDED to delete the old Security Context after OSCORE verification and verification of access rights succeed. The RS MUST delete the Security Context if it deletes the access token associated to it.
+          If the exchange was an update of access rights, i.e., a new Security Context was derived from a client that already had a Security Context in place, the is RECOMMENDED to delete the old Security Context after OSCORE verification and verification of access rights succeed. The RS MUST delete the Security Context if it deletes the access token associated to it.
         </t>
 
       </section> <!-- ends from line 481-->
 
       <section anchor="tok-ver" title="Access rights verification">
 
         <t>
-          The RS MUST follow the procedures defined in section 5.8.2 of <xref target="I-D.ietf-ace-oauth-authz"/>: if an RS receives an OSCORE-protected request from a client, then the RS processes it according to <xref target="I-D.ietf-core-object-security"/>. If OSCORE verification succeeds, and the target resource
-          requires authorization, the RS retrieves the authorization information from the access token associated to the Security Context. The RS then MUST verify that the authorization information covers the resource and the action requested.
+          The RS MUST follow the procedures defined in section 5.8.2 of <xref target="I-D.ietf-ace-oauth-authz"/>: if an RS receives an OSCORE-protected request from a client, then the RS processes it according to <xref target="RFC8613"/>. If OSCORE verification succeeds, and the target resource
+          requires authorization, the RS retrieves the authorization information using the access token associated to the Security Context. The RS then must verify that the authorization information covers the resource and the action requested.
         </t>
 
         <t>
-         The response code MUST be 4.01 (Unauthorized) in case the client has
-         not used the Security Context associated with the access token, or if RS has no valid access token for the client.  If RS has an access token for the client but
-         not for the resource that was requested, RS MUST reject the request
-         with a 4.03 (Forbidden).  If RS has an access token for the client
-         but it does not cover the action that was requested on the resource,
-         RS MUST reject the request with a 4.05 (Method Not Allowed).
+         The response code must be 4.01 (Unauthorized) in case the client has a valid token associated with that Security Context, but the Security Context has not been used before, as the proof-of-possession in this profile is performed by both parties verifying that they have established the same Security Context.
```

Comment:
I'm not 100% sure I follow what's going on here anymore where we talk about the security context "has not been used before", and why we are only mentioning 4.01 and not the 4.03 and 4.05 that were in the previous text.  Sorry for my bad memory :(

Reply:
No worries, totally understandable, it took me a while :)

So the "has not been used before" was my attempt at clarifying point 77. : 

> > 77.   The response code MUST be 4.01 (Unauthorized) in case the client has
> >        not used the Security Context associated with the access token, or if
> >        RS has no valid access token for the client.  If RS has an access
> >     
> >    The causality here seems a bit weird; how would the RS know what token was
> >    intended just from an OSCORE request that's using a random or different
> >    security context?
> > 
> > FP: The RS needs to keep this association between Security Context and token. This first section covers the case where the client is using a sec ctx not associated to any token or where that sec ctx was not used before (so the full OSCORE setup did not go through). Different (“valid”) security context associated to a token is covered in the next section. Is there anything I am missing here?
> 
> I think I'm just complaining about the wording: if a request comes in using
> some other (wrong) security context, how does the RS know what the relevant
> "the access token" is for that request?

And removing 4.03 and 4.05 was for removing duplicated framework text (commit 6dc9f4d6), as you suggested.

Reply:
Looking at the old text (in the -08) more carefully, I think the "has not used the Security Context associated with the token" is more in the sense of "used some other, different, security context".  The new text with "has not been used before" doesn't give me that same impression.